src/perms.rs

```diff
@@ -10,6 +10,16 @@ pub fn drop_privileges(user_uid: u32) {
     seteuid(user_uid.as_raw());
 }
 
+pub fn raise_privileges(resgid: ResGid, resuid: ResUid) {
+    let root_user = User::from_uid(resuid.real).unwrap().unwrap();
```

Comment:
The `real` UID isn't really the root user but the unprivileged invoking user.

Reply:
Got it. So should I use `resuid.effective` instead of `resuid.real`?

Reply:
Yes exactly. To be absolute sure I'd even pass a literal `0` here.

Actually also the interface of this `raise_privileges` function is a bit strange. Passing in the `ResGid` and `ResUid` types is confusing. Which one of real/effective/saved is used for restoring now? I think the function should just take exactly the UID and GID to raise privileges to, or even no parameters at all, and always change the effective UID/GID and groups to the one of literal UID 0 and GID 0.

Reply:
> Yes exactly. To be absolute sure I'd even pass a literal `0` here.
> 
> Actually also the interface of this `raise_privileges` function is a bit strange. Passing in the `ResGid` and `ResUid` types is confusing. Which one of real/effective/saved is used for restoring now? I think the function should just take exactly the UID and GID to raise privileges to, or even no parameters at all, and always change the effective UID/GID and groups to the one of literal UID 0 and GID 0.

👍 Done. https://github.com/waycrate/swhkd/commit/04e51255a4ef73151eac752c8c10dd51f1f30639

```rs
pub fn raise_privileges() {
    let root_user = User::from_uid(Uid::from_raw(0)).unwrap().unwrap();

    set_egid(0);
    set_euid(0);
    set_initgroups(&root_user, 0);
}
```

Reply:
Yes this should to nicely.

Reply:
@mgerstner Once again, thank you for the review ❤️